### PR TITLE
Add Location testing

### PR DESCRIPTION
**Pull request by abbymapes**
Added testing for creation of Location objects and calls to Geocoding and TimeZone APIs
- Tests wheter Location objects are constructed correctly (blank and valid locations)

Added timeZoneName property to location objects containing a human-readable name for time zone (Pacific Standard Time, rather than PST)
- Time Agent now outputs the full name, with abbreviation following, when a user makes a time zone request 
- Improves user experience (rather than abbreviations gives full name)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/utils/AgentUtils.java [hunk 1]
```diff
@@ -39,6 +39,10 @@ public static Output getOutput(QueryResult queryResult, String languageCode) {
     }
     fulfillment = fulfillment.equals("") ? "Can you repeat that?" : fulfillment;
```
Comment (aliu917) [inline portfolio/src/main/java/com/google/sps/utils/AgentUtils.java:40]: I think you can delete this line now that you added the following one.

File: portfolio/src/test/java/com/google/sps/LocationTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
+package com.google.sps.data;
+
+import com.google.maps.model.LatLng;
+import java.util.TimeZone;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
```
Comment (dibenede) [inline portfolio/src/test/java/com/google/sps/LocationTest.java:10]: delete or add comment

File: portfolio/src/test/java/com/google/sps/LocationTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+@RunWith(JUnit4.class)
+public final class LocationTest {
+  Location location = new Location("Los Angeles");
+  Location blankLocation = new Location(" ");
```
Comment (dibenede) [inline portfolio/src/test/java/com/google/sps/LocationTest.java:14]: should you be allowed to create a location with an invalid string?

File: portfolio/src/test/java/com/google/sps/LocationTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  public void checkLatitude() {
+    Double actual = location.getLat();
+    Double expected = 34.05223420;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void checkLongitude() {
+    Double actual = location.getLng();
+    Double expected = -118.24368490;
```
Comment (dibenede) [inline portfolio/src/test/java/com/google/sps/LocationTest.java:40]: is there any bounds checking on lat/long?

File: portfolio/src/main/java/com/google/sps/agents/Time.java [hunk 5]
```diff
@@ -137,10 +137,12 @@ public ZonedDateTime getCurrentTime(String locationName) {
 
   public String getZone(String locationName) {
     String timeZone = null;
+    Location place = new Location(locationName);
     ZonedDateTime time = getCurrentTime(locationName);
     DateTimeFormatter formatter = DateTimeFormatter.ofPattern("z");
     timeZone = time.format(formatter);
-    return timeZone;
+    String formattedZone = place.getTimeZoneName() + " (" + timeZone + ")";
+    return formattedZone;
```
Comment (dibenede) [inline portfolio/src/main/java/com/google/sps/agents/Time.java:145]: better to just return the concatenation than create the intermediate variable

File: portfolio/src/test/java/com/google/sps/LocationTest.java [hunk 6]
```diff
@@ -55,6 +55,13 @@ public void checkTimeZoneID() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void checkTimeZoneName() {
+    String actual = location.getTimeZoneName();
+    String expected = "Pacific Standard Time";
```
Comment (dibenede) [inline portfolio/src/test/java/com/google/sps/LocationTest.java:61]: What happens to this test when it's PDT?

File: portfolio/src/test/java/com/google/sps/LocationTest.java (hunk 3 above)
Comment (abbymapes) [reply portfolio/src/test/java/com/google/sps/LocationTest.java:14]: I changed it in next commit to make this purpose clearer. This case is when the user inputted location is not recognized by Geocoding API (i.e. a jibberish location, "home", "los flangelous" etc)

File: portfolio/src/test/java/com/google/sps/LocationTest.java (hunk 6 above)
Comment (abbymapes) [reply portfolio/src/test/java/com/google/sps/LocationTest.java:61]: The standard time is always returned by the TimeZone java class.

File: portfolio/src/test/java/com/google/sps/LocationTest.java (hunk 4 above)
Comment (abbymapes) [reply portfolio/src/test/java/com/google/sps/LocationTest.java:40]: The LatLng object has bounds on inputs for latitude and longitude. If the latitude input isn't in [-90,90], then the LatLng object automatically sets it to -90 (if it is less than -90) or 90 (if it is greater than 90). This is the same for longitude with [-180, 180]. So, for creating the coordinate object, the bounds are taken care of. I will add similar code to take care of this for the individual lat and lng properties.

**Pull request merged**